src/main/scala/sigmastate/Values.scala

```diff
@@ -420,13 +420,17 @@ object Values {
     def deconstruct = immutable.Seq[Any](elementType) ++ items
 
     def reconstruct(cs: immutable.Seq[Any]) = cs match {
-      case Seq(t: SType, vs@_*) => ConcreteCollection[SType](vs.asInstanceOf[Seq[Value[V]]].toIndexedSeq)(t)
+      case Seq(t: SType, vs@_*) => new ConcreteCollection[SType](vs.asInstanceOf[Seq[Value[V]]].toIndexedSeq, t)
       case _ =>
         illegalArgs("ConcreteCollection", "(IndexedSeq, SType)", cs)
     }
   }
   object ConcreteCollection {
-    def apply[V <: SType](items: Value[V]*)(implicit tV: V) = new ConcreteCollection(items.toIndexedSeq)
+    def apply[V <: SType](items: Value[V]*)(implicit tV: V) =
+      new ConcreteCollection(items.toIndexedSeq, tV)
+
+    def apply[V <: SType](items: => Seq[Value[V]])(implicit tV: V) =
```

Comment:
Why it is by-name?

Reply:
To trick the compiler and not have "same type after erasure" error with first `apply`. Using `IndexedSeq` triggers the same error with compiler generated "default" `apply` for case class. Kinda ugly, but did not find anything better.